### PR TITLE
test: [IV] Functional tests for IV backend function

### PR DESCRIPTION
**Pull request by komalg1**
## Purpose

* This PR adds functional tests for Integrated Vectorization backend function
* Will add Backend API IV functional tests in other PR

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

* Test the code

* Run unit tests
* Run functional tests
* Deploy to Azure

## What to Check
Verify that the following are valid
* All tests passing
* Admin + webapp working

### CODE REVIEW

File: code/tests/functional/conftest.py [hunk 1]
```diff
@@ -250,6 +250,53 @@ def setup_default_mocking(httpserver: HTTPServer, app_config: AppConfig):
         }
     )
 
+    httpserver.expect_request(
+        f"/datasources('{app_config.get('AZURE_SEARCH_DATASOURCE_NAME')}')",  # ?api-version=2023-10-01-Preview",
```
Comment (adamdougal) [inline code/tests/functional/conftest.py:254]:
```suggestion
        f"/datasources('{app_config.get('AZURE_SEARCH_DATASOURCE_NAME')}')",
```

File: code/tests/functional/tests/functions/advanced_image_processing/conftest.py [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
+import sys
 import pytest
 from tests.functional.app_config import AppConfig
-from backend.batch.utilities.helpers.config.config_helper import ConfigHelper
-from backend.batch.utilities.helpers.env_helper import EnvHelper
+
+sys.path.append(
+    os.path.join(os.path.dirname(sys.path[0]), "..", "..", "..", "backend", "batch")
+)
+from utilities.helpers.config.config_helper import ConfigHelper  # noqa: E402
```
Comment (adamdougal) [inline code/tests/functional/tests/functions/advanced_image_processing/conftest.py:10]: Does this still complain in the IDE? I think it's probably worth adding a comment explaining that it's required due to the funky sys path stuff we have to do. Same for the int vec conftest.

File: code/tests/functional/tests/functions/integrated_vectorization/test_integrated_vectorization_resource_creation.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+                },
+                "dataVersion": "",
+                "metadataVersion": "1",
+                "eventTime": "2024-05-10T09:22:51.5565464Z",
+            }
+        )
+    )
+
+
+def test_config_file_is_retrieved_from_storage(
```
Comment (adamdougal) [inline code/tests/functional/tests/functions/integrated_vectorization/test_integrated_vectorization_resource_creation.py:64]: These tests are great!

great

File: code/tests/functional/tests/functions/integrated_vectorization/test_integrated_vectorization_resource_creation.py [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+    verify_request_made(
+        mock_httpserver=httpserver,
+        request_matcher=RequestMatcher(
+            path=f"/datasources('{app_config.get('AZURE_SEARCH_DATASOURCE_NAME')}')",
+            method="PUT",
+            json={
+                "name": app_config.get("AZURE_SEARCH_DATASOURCE_NAME"),
+                "type": "azureblob",
+                "credentials": {
+                    "connectionString": f"DefaultEndpointsProtocol=https;AccountName={app_config.get('AZURE_BLOB_ACCOUNT_NAME')};AccountKey=c29tZS1ibG9iLWFjY291bnQta2V5;EndpointSuffix=core.windows.net"
```
Comment (cecheta) [inline code/tests/functional/tests/functions/integrated_vectorization/test_integrated_vectorization_resource_creation.py:100]:
```suggestion
                    "connectionString": f"DefaultEndpointsProtocol=https;AccountName={app_config.get('AZURE_BLOB_ACCOUNT_NAME')};AccountKey={app_config.get('AZURE_SEARCH_DATASOURCE_NAME')};EndpointSuffix=core.windows.net"
```

File: code/tests/functional/tests/functions/advanced_image_processing/conftest.py (hunk 2 above)
Comment (komalg1) [reply code/tests/functional/tests/functions/advanced_image_processing/conftest.py:11]: Added a comment, let me know if it conveys the message

**Review by adamdougal — APPROVED** [review]
Great job!

**Pull request merged**